net/instaweb/rewriter/csp_test.cc

```diff
@@ -0,0 +1,135 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Author: morlovich@google.com (Maksim Orlovich)
+
+#include "net/instaweb/rewriter/public/csp.h"
+
+#include <memory>
+
+#include "pagespeed/kernel/base/basictypes.h"
+#include "pagespeed/kernel/base/gtest.h"
+#include "pagespeed/kernel/base/string.h"
+#include "pagespeed/kernel/base/string_util.h"
+
+namespace net_instaweb {
+
+namespace {
+
+TEST(CspParseSourceTest, Quoted) {
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kSelf),
+      CspSourceExpression::Parse("'self' "));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kSelf),
+      CspSourceExpression::Parse("   'sElf' "));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kStrictDynamic),
+      CspSourceExpression::Parse("   'strict-dynamic' "));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kUnsafeInline),
+      CspSourceExpression::Parse("'unsafe-inline'"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kUnsafeEval),
+      CspSourceExpression::Parse("'unsafe-eval'"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kUnsafeHashedAttributes),
+      CspSourceExpression::Parse("'unsafe-hashed-attribUtes'"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kUnknown),
+      CspSourceExpression::Parse("'nonce-qwertyu12345'"));
+}
+
+TEST(CspParseSourceTest, NonQuoted) {
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kUnknown),
+      CspSourceExpression::Parse("   "));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kSchemeSource, "https:"),
+      CspSourceExpression::Parse(" https:"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kSchemeSource,
+                          "weird-schema+-1.0:"),
+      CspSourceExpression::Parse("weird-schema+-1.0:"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kHostSource, "*.example.com"),
+      CspSourceExpression::Parse("*.example.com"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kHostSource,
+                          "http://www.example.com/dir"),
+      CspSourceExpression::Parse("http://www.example.com/dir"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kHostSource,
+                          "http://www.example.com/dir/file.js"),
+      CspSourceExpression::Parse("http://www.example.com/dir/file.js"));
+
+  EXPECT_EQ(
+      CspSourceExpression(CspSourceExpression::kHostSource, "*"),
+      CspSourceExpression::Parse("*"));
+}
+
+TEST(CspParseTest, Empty) {
+  std::unique_ptr<CspPolicy> policy(CspPolicy::Parse(
+      "   "));
+  EXPECT_EQ(policy, nullptr);
+}
+
+TEST(CspParseTest, Basic) {
+  std::unique_ptr<CspPolicy> policy(CspPolicy::Parse(
+    "default-src *; script-src 'unsafe-inline' 'unsafe-eval'"));
+  ASSERT_NE(policy, nullptr);
```

Comment:
I find error messages quicker to grok when you spell this:
   ASSERT_TRUE(policy != nullptr);
but this is fine too.